src/lib/files/ProjectFilesManager.ts

```diff
@@ -0,0 +1,75 @@
+import { createReadStream, lstat, readFileSync, statSync } from 'fs-extra'
+import * as glob from 'globby'
+import { join } from 'path'
+import { reject } from 'ramda'
+
+export const pathToFileObject = (root: string, prefix: string = '') => {
+  return (path: string): BatchStream => {
+    const realAbsolutePath = join(root, path)
+    const stats = statSync(realAbsolutePath)
+    return {
+      path: join(prefix, path),
+      content: createReadStream(realAbsolutePath),
+      byteSize: stats.size,
+    }
+  }
+}
+
+export class ProjectFilesManager {
+  public static DEFAULT_IGNORED_FILES = [
```

Comment:
Do you think it makes sense to add `readonly` or use the `get` accessor, without letting anyone set this variable?